### PR TITLE
Handle PropfindError exception

### LINKED ISSUE #748: Catch and return useful error messages for propfinderrors
Sentry error: https://thunderbird.sentry.io/issues/6032781387/?project=4505428124827648&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=14d&stream_index=3
https://thunderbird.sentry.io/issues/6069130961/?project=4505428124827648&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=14d&stream_index=2

We should catch all instances of PropfindError and a useful error message like "Server could not be reached, please ensure your CalDAV settings are correct."

### PR DESCRIPTION
**Pull request by devmount**
## Description of the Change

This adds exception handling for caldav PropfindErrors.

## Applicable Issues
Fixes #748

### CODE REVIEW

**Review by devmount — COMMENTED** [review]
I moved some comments above the corresponding exception, hope that's ok. I find it more readable like this.

File: backend/src/appointment/controller/calendar.py [hunk 1]
```diff
@@ -310,15 +310,21 @@ def test_connection(self) -> bool:
...
+        # Max retries exceeded, bad connection, missing schema, etc...
+        except requests.exceptions.RequestException as ex:
             return False
-        except caldav.lib.error.NotFoundError as ex:  # Good server, bad url.
+        # Good server, bad url.
+        except caldav.lib.error.NotFoundError as ex:
+            return False
+        # Some properties could not be retrieved.
+        except caldav.lib.error.PropfindError as ex:
             return False
```
[the remark this answers is not in the document]
Comment (MelissaAutumn) [reply backend/src/appointment/controller/calendar.py:328]: Apparentally we can tuple these up onto one line. TIL. 

So maybe:
```suggestion
        """
        RequestException: Max retries exceeded, bad connection, missing schema, etc...
        NotFoundError: Good server, bad url.
        PropfindError: Some properties could not be retrieved.
        """
        except (requests.exceptions.RequestException, caldav.lib.error.NotFoundError, caldav.lib.error.PropfindError) as ex:
            return False
```
(might want to confirm this is working as expected though lol)

**Review by MelissaAutumn — APPROVED** [review]
Thanks!

File: backend/src/appointment/controller/calendar.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (devmount) [reply backend/src/appointment/controller/calendar.py:328]: Oh nice, will test that 👍🏻

[the remark this answers is not in the document]
Comment (devmount) [reply backend/src/appointment/controller/calendar.py]: Mh, this gives an invalid syntax error, seems like you cannont indent multiline comments like this 😅 We could move the comments one line below though. I'll try something.

**Pull request merged**